### DISCUSSION TITLE
New `remove` API for builder

### DISCUSSION THREAD

**Discussion by cf-coheretech** [fmeum/with_cfg.bzl#136] [Ideas]
Is it possible to add an inverse API to `extend`?

Something like:
remove(setting, value): `value` is removed from the current of entries of setting if `value` is present. value has to evaluate to a scalar of the same type as the entries of setting (usually a string or a select expression evaluating to a string). This is useful for repeatable settings such as copt.

The specific problem I'm trying to solve is that I want to instrument only some of my codebase for coverage. The instrumentation_filter flag does not seem to do anything for C++ code (https://github.com/bazelbuild/bazel/issues/15627). The `-fno-profile-instr-generate` is ignored for linking, so the only solution I can think of is to remove the `-fprofile-instr-generate` flag with a transition.

**Comment by fmeum** [comment]
Happy to add this feature if necessary. If we do that, I would prefer `value` to be a sequence so that you could also remove flags that only make sense in sequence (say `-Xlinker -whatever`).

That said, for your concrete use case, have you tried `extend`ing `features` with `-coverage`? If that works, I would consider that a cleaner solution.

**Reply by cf-coheretech** [reply]
Yes, I realized I can use `set("collect_code_coverage", "false")` for my use case
